src/client/index.js

```diff
@@ -0,0 +1,188 @@
+import hooks from './hooks';
+import utils from './utils';
+
+const defaults = {
+  usernameField: 'email',
+  passwordField: 'password',
+  userEndpoint: '/users',
+  localEndpoint: '/auth/local',
+  tokenEndpoint: '/auth/token'
+};
+
+export default function(options = {}) {
+  const authOptions = Object.assign({}, defaults, options);
+
+  return function() {
+    const app = this;
+
+    app.authenticate = function(options) {
+      if (!options.type) {
+        throw new Error('You need to provide a `type` attribute when calling app.authenticate()');
+      }
+      
+      let endPoint;
+
+      if (options.type === 'local') {
+        endPoint = authOptions.localEndpoint;
+      } else if (options.type === 'token') {
+        endPoint = authOptions.tokenEndpoint;
+      }
+      else {
+        throw new Error(`Unsupported authentication 'type': ${options.type}`);
+      }
+
+      // return new Promise(function(resolve, reject) {
```

Comment:
That's the promisfying we were trying to do the other night so that a developer doesn't have to call `app.io.on('connect', cb)` before they can authenticate. We would handle that for them. I couldn't get it to work. Too many promises and event listeners.